Mathlib/Tactic/PushNeg.lean

```diff
@@ -55,6 +55,7 @@ def transformNegationStep (e : Expr) : SimpM (Option Simp.Step) := do
     catch _ => return none
   let e_whnf ← whnfR e
   let some ex := e_whnf.not? | return Simp.Step.visit { expr := e }
+  let ex := ex.consumeMData
   match ex.getAppFnArgs with
```

Comment:
There's a function to cleanup annotations (like `outParam` etc.) and metadata all at once
```suggestion
  match ex.cleanupAnnotations.getAppFnArgs with
```

Please also make sure to add Yury's examples from Zulip into the test file. Once these are done (and you fix the errors in the rest of mathlib, and the fixes are not regressions) I'm happy with this change to `push_neg`.

Reply:
Kyle, thank you!  I have pushed fixes for all the issues that were brought up: they all appear improvements to me!

I will wait to see how CI deals with the current version and, once it is green, I will push your fix.